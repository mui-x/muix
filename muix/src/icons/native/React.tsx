import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)