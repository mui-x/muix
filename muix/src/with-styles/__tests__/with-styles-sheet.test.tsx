/** @jsx createElement */
import React from 'react'
import { createElement, toClassNamesWithQuery } from 'reactxx-sheeter'
import { TSheeter, TComponents } from 'reactxx-typings'

import { initPlatform, Shape, ts, shallow, mount } from "reactxx-tests";

export const createSheet = () =>
  (ts.sheet = {
    root: ts.view = {},
    label: ts.text = {},
    webOnly: ts.web = {},
    nativeOnly: ts.nativeText = {},
  });

describe("WITH STYLES", () => {
  it("WEB", () => {
    initPlatform(true);
    const comp = shallow(<div
      classNameX={toClassNamesWithQuery(null, { color: 'red' })}
      //styleX={{ margin: 10, padding: 20 }]}
    >
      Hallo world
    </div>);
    expect(comp).toMatchSnapshot();
    expect(comp.props().className).toMatchSnapshot()
  });
  const ctx = React.createContext(123)
  it("WEB", () => {
    initPlatform(true);
    let comp = mount(<ctx.Provider value={1}>
      <React.Fragment>
        <h1>Hallo</h1>
        <div>
          <ctx.Consumer>
            {num => <div id={num.toString()} />}
          </ctx.Consumer>
        </div>
      </React.Fragment>
    </ctx.Provider>)
    expect(comp).toMatchSnapshot();
    comp = mount(<React.Fragment>
      <h1>Hallo</h1>
      <div>
        <ctx.Consumer>
          {num => <div id={num.toString()} />}
        </ctx.Consumer>
      </div>
    </React.Fragment>
    )
    expect(comp).toMatchSnapshot();
  });
});
