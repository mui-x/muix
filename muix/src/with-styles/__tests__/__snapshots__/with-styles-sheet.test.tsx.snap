// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLES WEB 1`] = `
<div
  className="a"
  style={
    Object {
      "margin": 10,
      "padding": 20,
    }
  }
>
  Hallo world
</div>
`;

exports[`WITH STYLES WEB 2`] = `"a"`;

exports[`WITH STYLES WEB 3`] = `
Array [
  <h1>
    Hallo
  </h1>,
  <div>
    <div
      id="1"
    />
  </div>,
]
`;

exports[`WITH STYLES WEB 4`] = `
Array [
  <h1>
    Hallo
  </h1>,
  <div>
    <div
      id="123"
    />
  </div>,
]
`;
